### PR TITLE
Simplify functions in preparation for PartiQL-PATH

### PR DESCRIPTION
**Pull request by RCHowell**
## Relevant Issues

#1496 

## Description

This PR simplifies functions in SPI in preparation for the SQL-PATH. This is being done in conjunction with the PartiQL-Environment where connectors will be implemented on top of the catalog interfaces. This is an intermediate step towards the final state.

## Other Information
- Updated Unreleased Section in CHANGELOG: **No**
- Any backward-incompatible changes? **No**
- Any new external dependencies? **No**
- Do your changes comply with the [Contributing Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CONTRIBUTING.md)
  and [Code Style Guidelines](https://github.com/partiql/partiql-lang-kotlin/blob/main/CODE_STYLE.md)? **Yes**

## License Information

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.

### CODE REVIEW

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/internal/Accumulator.kt [hunk 1]
```diff
@@ -130,7 +128,7 @@ internal fun checkIsBooleanType(funcName: String, value: PartiQLValue) {
 }
 
 @OptIn(PartiQLValueExperimental::class)
-internal fun PartiQLValue.isUnknown(): Boolean = this.type == PartiQLValueType.MISSING || this.isNull
+internal fun PartiQLValue.isUnknown(): Boolean = this.type == PartiQLValueType.ANY || this.isNull
```
Comment (johnedquinn) [inline partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/internal/Accumulator.kt:131]: Mental note

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnIsNull.kt [hunk 2]
```diff
@@ -28,7 +26,7 @@ internal object Fn_IS_NULL__ANY__BOOL : Fn {
     )
 
     override fun invoke(args: Array<PartiQLValue>): PartiQLValue {
-        if (args[0].type == MISSING) {
+        if (args[0].type == ANY) {
```
Comment (johnedquinn) [inline partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnIsNull.kt:29]: Mental note

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnEq.kt [hunk 3]
```diff
@@ -49,7 +47,7 @@ internal object Fn_EQ__ANY_ANY__BOOL : Fn {
     override fun invoke(args: Array<PartiQLValue>): PartiQLValue {
         val lhs = args[0]
         val rhs = args[1]
-        if (lhs.type == MISSING || rhs.type == MISSING) {
+        if (lhs.type == ANY || rhs.type == ANY) {
```
Comment (johnedquinn) [inline partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnEq.kt:50]: Why ANY?

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnIsNull.kt (hunk 2 above)
Comment (johnedquinn) [reply partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnIsNull.kt:29]: Why ANY?

**Review by johnedquinn — CHANGES_REQUESTED** [review]
There are 31 conformance tests now failing -- potentially due to the use of ANY here.

**Comment by RCHowell** [comment]
I likely just pulled the Fns from an older v1 commit. Let me get the latest definitions.

File: partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnEq.kt (hunk 3 above)
Comment (RCHowell) [reply partiql-planner/src/main/kotlin/org/partiql/planner/internal/fn/builtins/FnEq.kt:50]: Update, these were automatically changed by IntelliJ due to the "replaceWith" on the MISSING deprecated annotation.

**Review by johnedquinn — APPROVED** [review]

**Pull request merged**